src/main/webapp/script.js

```diff
@@ -21,11 +21,19 @@ $(document).ready(() => {
   // Set up trigger to add hidden POI form elements upon submission.
   addHiddenPoiFormTrigger();
 
-  isSignedIn().then((signInStatus) => {
+  getAuthObject().then((authObject) => {
     // Display the sign-in page or "start trip" form depending on sign in status.
-    if (signInStatus) {
+    if (authObject.email !== null) {
```

Comment:
After logging out I still see the trip creation page. I think the 'hasOwnProperty' or 'in' operators would do what you want instead of !== null.  Checking for null/non-existence in javascript is fiddly :/

Reply:
`hasOwnProperty` returns `true` if the property is present, but set to `null`; I didn't see anything on using `in` to check for `null` values in JavaScript? I removed the `loggedIn` boolean property because Bootsie and Adam argued it was unnecessary, as I could check for `null` here -- what would you advise I do?

Reply:
Actually, `hasOwnProperty` works! I didn't realize that the `email` object isn't present; in the backend, when the property is set to `null` and the object is converted to JSON through GSON, the property disappears. I just updated this, I think it should work now -- thanks for verifying this.